controllers/provisioning_controller.go

```diff
@@ -22,15 +22,32 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	osconfigv1 "github.com/openshift/api/config/v1"
 	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
 )
 
+const (
+	componentNamespace = "openshift-machine-api"
+	componentName      = "cluster-baremetal-operator"
+)
+
 // ProvisioningReconciler reconciles a Provisioning object
 type ProvisioningReconciler struct {
-	client.Client
-	Log    logr.Logger
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	Client client.Client
 	Scheme *runtime.Scheme
+	Log    logr.Logger
```

Comment:
Does the logger need to be public? Is anything outside of the reconciler going to use it?

Reply:
It was put into Reconciler by kubebuilder. Just the formatting changed in this PR. I'll leave a TODO here for us to see if there is a reason why it needs to be part of the Reconciler struct.